sgkit/model.py

```diff
@@ -156,5 +159,24 @@ def create_genotype_dosage_dataset(
     }
     if variant_id is not None:
         data_vars["variant_id"] = ([DIM_VARIANT], variant_id)
-    attrs: Dict[Hashable, Any] = {"contigs": variant_contig_names}
+    attrs: Dict[Hashable, Any] = {
+        "contigs": variant_contig_names,
+        "source": f"sgkit-{sg.__version__}",
+    }
     return create_dataset(data_vars=data_vars, attrs=attrs)
+
+
+def num_contigs(ds: xr.Dataset) -> ArrayLike:
+    """Return the number of contigs in a dataset."""
+    if "contigs" in ds.attrs:
```

Comment:
Could we add a deprecation warning here? It will only be triggered by internal code but would still communicate the change to users.

Reply:
Goode idea - I've added a warning in the latest update